An error raised in assignment 3

**The error occurded in the following block (This blocked is designed to verify the cliping method)**：

```
# Run this cell without modification

print("Now let's see what happen if gradient clipping is not enable!\n")
try:
    failed_mlp_trainer, _ = run(MLPTrainer, dict(
        max_iteration=3000,
        evaluate_interval=100, 
        parameter_std=0.01,
        learning_rate=0.001,
        hidden_dim=100,
        clip_gradient=False,  # <<< Gradient clipping is OFF!
        env_name="CartPole-v0"
    ), reward_threshold=195.0)
    print("\nWe expect to see bad performance (<195). "
          "The performance without gradient clipping: {}."
          "".format(failed_mlp_trainer.evaluate()))
except AssertionError as e:
    print(traceback.format_exc())
    print("Infinity happen during training. It's OK since the gradient is not bounded.")
finally:
    print("Try next cell to see the impact of gradient clipping.")
```
**Details about the error raised (The fllowing contents are printed out after running the testing block)**：
```
Now let's see what happen if gradient clipping is not enable!

(0.0s,+0.0s)	Iteration 0, current mean episode reward is 9.04. 
d:\programdata\anaconda3\envs\ierg5350\lib\site-packages\ipykernel_launcher.py:70: RuntimeWarning: overflow encountered in double_scalars
d:\programdata\anaconda3\envs\ierg5350\lib\site-packages\ipykernel_launcher.py:70: RuntimeWarning: invalid value encountered in double_scalars
Try next cell to see the impact of gradient clipping.
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_23620/1928379951.py in <module>
     12         env_name="CartPole-v0"
---> 13     ), reward_threshold=195.0)
     14     print("\nWe expect to see bad performance (<195). "

~\AppData\Local\Temp/ipykernel_23620/2935914240.py in run(trainer_cls, config, reward_threshold)
     16     for i in range(config['max_iteration'] + 1):
---> 17         stat = trainer.train()
     18         stats.append(stat or {})

~\AppData\Local\Temp/ipykernel_23620/1932366688.py in train(self)
     83                 gradient = self.compute_gradient(
---> 84                     processed_states, actions, rewards, tau, T
     85                 )

~\AppData\Local\Temp/ipykernel_23620/1566270218.py in compute_gradient(self, processed_states, actions, rewards, tau, T)
     87             "Invalid value occurs in output_gradient! {}".format(
---> 88                 output_gradient)
     89         assert np.all(np.isfinite(hidden_gradient)), \

AssertionError: Invalid value occurs in output_gradient! [[ 0.00000000e+000  6.76629673e+307]
 [ 0.00000000e+000 -4.37202797e+307]
 [ 0.00000000e+000  1.35839278e+308]
 [ 0.00000000e+000 -1.04095533e+308]
 [ 0.00000000e+000  8.26476911e+307]
 [ 0.00000000e+000  4.78122982e+307]
 [ 0.00000000e+000  1.43548055e+307]
 [ 0.00000000e+000             -inf]
 [ 0.00000000e+000 -4.42654726e+307]
 [ 0.00000000e+000             -inf]
 [ 0.00000000e+000 -1.34577615e+308]
 [ 0.00000000e+000  1.77084445e+308]
 [ 0.00000000e+000 -1.05785186e+307]
 [ 0.00000000e+000  5.43210268e+307]
 [ 0.00000000e+000  2.35034865e+307]
 [ 0.00000000e+000  1.18841978e+308]
 [ 0.00000000e+000 -1.07229034e+308]
 [ 0.00000000e+000  6.05203091e+307]
 [ 0.00000000e+000 -7.51744290e+307]
 [ 0.00000000e+000 -8.02180137e+307]
 [ 0.00000000e+000  9.66273904e+307]
 [ 0.00000000e+000 -1.33247518e+307]
 [ 0.00000000e+000 -6.92675394e+307]
 [ 0.00000000e+000 -5.87569807e+306]
 [ 0.00000000e+000  4.52424867e+307]
 [ 0.00000000e+000  4.08847468e+307]
 [ 0.00000000e+000 -5.53828357e+307]
 [ 0.00000000e+000 -1.25418973e+308]
 [ 0.00000000e+000  1.47403666e+308]
 [ 0.00000000e+000 -5.93074557e+306]
 [ 0.00000000e+000  1.23342194e+308]
 [ 0.00000000e+000 -4.64790548e+306]
 [ 0.00000000e+000 -8.09147379e+307]
 [ 0.00000000e+000  1.16011873e+308]
 [ 0.00000000e+000  1.32838101e+308]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000  2.41169773e+307]
 [ 0.00000000e+000             -inf]
 [ 0.00000000e+000 -7.27871333e+307]
 [ 0.00000000e+000 -8.10853140e+307]
 [ 0.00000000e+000 -9.00542639e+307]
 [ 0.00000000e+000  9.57023865e+307]
 [ 0.00000000e+000  8.19198868e+307]
 [ 0.00000000e+000             -inf]
 [ 0.00000000e+000 -3.94767584e+307]
 [ 0.00000000e+000  9.38175072e+307]
 [ 0.00000000e+000 -1.64544176e+308]
 [ 0.00000000e+000             -inf]
 [ 0.00000000e+000  1.17229054e+308]
 [ 0.00000000e+000  1.01281822e+308]
 [ 0.00000000e+000 -4.01068912e+307]
 [ 0.00000000e+000 -1.25940721e+308]
 [ 0.00000000e+000  1.01535147e+308]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000  9.58972617e+306]
 [ 0.00000000e+000 -1.65206668e+308]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000  4.29136349e+307]
 [ 0.00000000e+000 -1.04845760e+308]
 [ 0.00000000e+000 -1.12420497e+308]
 [ 0.00000000e+000  9.45725689e+307]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000  4.57456454e+307]
 [ 0.00000000e+000 -1.53276978e+308]
 [ 0.00000000e+000  1.56072276e+308]
 [ 0.00000000e+000  1.62737021e+308]
 [ 0.00000000e+000 -9.44397010e+307]
 [ 0.00000000e+000  1.50003077e+308]
 [ 0.00000000e+000  2.02011409e+307]
 [ 0.00000000e+000 -1.21159488e+306]
 [ 0.00000000e+000  5.02540205e+307]
 [ 0.00000000e+000 -1.00005611e+308]
 [ 0.00000000e+000  6.63077482e+307]
 [ 0.00000000e+000  7.28808289e+307]
 [ 0.00000000e+000 -5.33929978e+307]
 [ 0.00000000e+000 -1.43867299e+307]
 [ 0.00000000e+000 -3.88647021e+307]
 [ 0.00000000e+000 -4.39579593e+307]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000             -inf]
 [ 0.00000000e+000  1.11382477e+308]
 [ 0.00000000e+000  8.85007990e+306]
 [ 0.00000000e+000 -5.78139456e+307]
 [ 0.00000000e+000 -4.43131683e+307]
 [ 0.00000000e+000  4.96153628e+307]
 [ 0.00000000e+000 -6.56681116e+307]
 [ 0.00000000e+000 -5.44928539e+307]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000 -7.26406369e+307]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000  3.57453705e+307]
 [ 0.00000000e+000  7.20987767e+307]
 [ 0.00000000e+000 -3.51639318e+307]
 [ 0.00000000e+000              inf]
 [ 0.00000000e+000 -8.77285555e+307]
 [ 0.00000000e+000  9.47702623e+307]
 [ 0.00000000e+000 -4.55382896e+307]
 [ 0.00000000e+000  6.00629438e+307]
 [ 0.00000000e+000  9.27333340e+307]]

During handling of the above exception, another exception occurred:

NameError                                 Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_23620/1928379951.py in <module>
     16           "".format(failed_mlp_trainer.evaluate()))
     17 except AssertionError as e:
---> 18     print(traceback.format_exc())
     19     print("Infinity happen during training. It's OK since the gradient is not bounded.")
     20 finally:

NameError: name 'traceback' is not defined
```

--- a/assignment3/utils.py
+++ b/assignment3/utils.py
@@ -1,9 +1,14 @@
 import copy
+import sys
 import time
+import traceback
 from collections import deque
 
 import gym
 import numpy as np
+
+traceback
+sys
 
 
 def wait(sleep=0.2):

